src/View/Helper/CategoriesHelper.php

```diff
@@ -0,0 +1,169 @@
+ */
+class CategoriesHelper extends Helper
+{
+    public $helpers = ['Form', 'Property'];
+
+    /**
+     * Control for categories
+     *
+     * @param string $name The name
+     * @param mixed|null $value The value
+     * @return string
+     */
+    public function control(string $name, $value): string
+    {
+        $options = ['label' => false];
+        if (!$this->isTree()) {
+            return sprintf(
+                '<div class="categories"><h3>%s</h3>%s</div>',
+                __('Global'),
+                $this->Property->control($name, $value, $options)
+            );
+        }
+
+        return $this->html($name, $value, $options);
+    }
+
+    /**
+     * Html for categories tree
+     *
+     * @param string $name The name
+     * @param mixed|null $value The value
+     * @param array $options The options
+     * @return string
+     */
+    public function html(string $name, $value, array $options): string
+    {
+        $html = '';
+        $tree = $this->tree();
+        $hiddenField = true; // hiddenField false on all controls except the first
+        foreach ($tree as $node) {
+            $html .= $this->node($node, $name, $value, $options, $hiddenField);
+        }
+
+        return $html;
+    }
+
+    /**
+     * Html for single node of categories tree
+     *
+     * @param array $node The category node
+     * @param string $name The name
+     * @param mixed|null $value The value
+     * @param array $options The options
+     * @param bool $hiddenField The hiddenField flag
+     * @return string
+     */
+    public function node(array $node, string $name, $value, array $options, bool &$hiddenField): string
+    {
+        $controlOptions = $this->controlOptions($node, $value, $options, $hiddenField);
+        $title = count(array_keys($controlOptions['options'])) === 1 ? __('Global') : $node['label'];
+
+        return sprintf(
+            '<div class="categories"><h3>%s</h3>%s</div>',
+            $title,
+            $this->Form->control($name, $controlOptions)
+        );
+    }
+
+    /**
+     * Control options for category node.
+     *
+     * @param array $node The category node
+     * @param mixed|null $value The value
+     * @param array $options The options
+     * @param bool $hiddenField The hiddenField flag
+     * @return array
+     */
+    public function controlOptions(array $node, $value, array $options, bool &$hiddenField): array
+    {
+        $controlOptions = $options + [
+            'type' => 'select',
+            'multiple' => 'checkbox',
+            'value' => (array)Hash::extract((array)$value, '{n}.name'),
+            'hiddenField' => $hiddenField,
+        ];
+        $hiddenField = false;
+        if (empty($node['children'])) {
+            $controlOptions['options'][0] = ['value' => $node['name'], 'text' => $node['label']];
+
+            return $controlOptions;
+        }
+        $key = 0;
+        foreach ($node['children'] as $child) {
```

Comment:
why not using `$node['children'] as $index => $child` instead of the `$key` counter?